notification-server/src/models.rs

```diff
@@ -1,29 +1,28 @@
 use concordium_rust_sdk::base::contracts_common::AccountAddress;
 use enum_iterator::Sequence;
-use num_bigint::BigInt;
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
 
 /// Represents details for a notification.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
 pub struct NotificationInformation {
     /// The blockchain account address unawarely involved in the notification
     /// emitting event.
-    pub address: AccountAddress,
+    #[serde(rename = "recipient")]
+    pub address:          AccountAddress,
     /// The amount being involved in the notification emitting event.
-    pub amount:  BigInt,
+    pub amount:           String,
```

Comment:
It is supposed to be in string anyways, so instead of dealing with serialization I rather do this. 

https://concordium.atlassian.net/wiki/spaces/EN/pages/1502019590/Message+data+payload

Reply:
We usually (for.. reasons) serialize amounts as string, but wouldn't it be more representative to use the actual amount types? The document you linked also states that the two types of notifications have different data associated with them.. I would imagine a datamodel such as

```rust
#[serde(tag = "type")]
enum NotificationData {
  #[serde(rename  = "ccd-tx")]
  CCD { amount: Amount, address: AccountAddress, ... },
  #[serde(rename  = "cis2-tx")]  
  CIS2 { amount: cis2::TokenAmount, ... },
}
```

Reply:
I see little value for this. 

This is not an enum and will never be either.

This is the contract from the notification-server to the client. That is not and should not be dependent on the concordium types

Reply:
Is there a guarantee that the serialisation of Amount is equivalent to the string amount?

I see little value in this and it may even be causing issues with the contract https://concordium.atlassian.net/wiki/spaces/EN/pages/1502019590/Message+data+payload

Reply:
The latter part of splitting it up into two types makes a lot of sense. Would prefer doing that when I get to CIS2 event specific data

Reply:
All this typed should be included into both types anways. :)

Reply:
Makes sense